x/genutil/genesis.go

```diff
@@ -16,9 +16,31 @@ func InitGenesis(
 	ctx context.Context, stakingKeeper types.StakingKeeper,
 	deliverTx genesis.TxHandler, genesisState types.GenesisState,
 	txEncodingConfig client.TxEncodingConfig,
-) (validators []abci.ValidatorUpdate, err error) {
+) (validatorUpdates []module.ValidatorUpdate, err error) {
 	if len(genesisState.GenTxs) > 0 {
-		validators, err = DeliverGenTxs(ctx, genesisState.GenTxs, stakingKeeper, deliverTx, txEncodingConfig)
+		cometValidatorUpdates, err := DeliverGenTxs(ctx, genesisState.GenTxs, stakingKeeper, deliverTx, txEncodingConfig)
+		if err != nil {
+			return nil, err
+		}
+		validatorUpdates = make([]module.ValidatorUpdate, len(cometValidatorUpdates))
+		for i, v := range cometValidatorUpdates {
+			if ed25519 := v.PubKey.GetEd25519(); len(ed25519) > 0 {
+				validatorUpdates[i] = module.ValidatorUpdate{
+					PubKey:     ed25519,
+					PubKeyType: "ed25519",
```

Comment:
we should abstract this not to duplicate?

these constants should be from the crypto package in order to avoid accidental mistakes

Reply:
let me see, i will do this

Reply:
done!